paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/RenameTagAction.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.flink.action;
+
+import org.apache.paimon.utils.StringUtils;
+
+import java.util.Map;
+
+/** Rename Tag action for Flink. */
+public class RenameTagAction extends TableActionBase {
+    private final String tagName;
+    private final String newTagName;
+
+    public RenameTagAction(
+            String warehouse,
+            String databaseName,
+            String tableName,
+            Map<String, String> catalogConfig,
+            String tagName,
+            String newTagName) {
+        super(warehouse, databaseName, tableName, catalogConfig);
+        this.tagName = tagName;
+        this.newTagName = newTagName;
+    }
+
+    @Override
+    public void run() throws Exception {
+        if (StringUtils.isEmpty(tagName) || StringUtils.isEmpty(newTagName)) {
+            throw new RuntimeException(String.format("tagName or newTagName can not be empty"));
+        }
```

Comment:
String.format("The specified tag name [%s] or target tag name [%s] cannot be empty.",  tagName, targetName)

Reply:
done